lib/function-loader.js

```diff
@@ -25,14 +26,19 @@ function isESM(filePath) {
   }
 
   // find the functions package.json and see if it has a type field
-  const functionPkg = require(path.join(pathParsed.dir, 'package.json'));
-  if (functionPkg.type === 'module') {
-    return true;
+  // look in the parent directory if the function is in a subdirectory
+  let dir = pathParsed.dir;
+  while (!hasPackageJson(dir)) {
+    if (dir === '/') return false;
+    dir = path.join(dir, '..');
   }
```

Comment:
Wouldn't this work?
```suggestion
  while (!hasPackageJson(dir)) {
    const d = path.join(dir, '..');
    if (d === dir) return false;
    dir = d
  }
```

Reply:
I have changed it to use `const rootDir = path.parse(process.cwd()).root;` which should work on all systems supported by Node.js